torch/_inductor/codegen/simd.py

```diff
@@ -1818,6 +1809,8 @@ def get_nd_tilings(
                 # Check if the variable ranges coincide with the node ranges.
                 # This lets us partition the variables into pointwise and reduction
                 # splits.
+                # FIXME right now we mishandle cases like [[], [8,8]]. In those cases, we
```

Comment:
TODO: fix this? Right now, the numels check will reject these bad tilings. But it seems like we should have a more principled way of handling `[]`

Reply:
I generalized this logic in [04b7a65](https://github.com/pytorch/pytorch/pull/137243/commits/04b7a65fae060cbde4a08eb9dbf32184c6be5063). Now, we can handle multiple variables in either the pointwise or reduction categories. The basic approach is to assume that the first N variables are pointwise, and solve for N by taking the cumulative product of their numels. I'm not sure if there are cases that violate this assumption, but at least for the test cases which need this analysis, we've always had pointwise vars prior to reduction ones.